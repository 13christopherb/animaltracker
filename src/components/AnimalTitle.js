import React, {Component} from "react";
import {Link} from 'react-router-dom';
import {connect} from 'react-redux';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'


class AnimalTitle extends React.Component {


    render() {
        return (
            <tr key={this.props.animal.id}>
                <td>
                    {this.props.animal.species}
                </td>
                <td>
                    {this.props.animal.name}
                </td>
                <td>
                    {this.props.animal.weight}
                </td>
                <td>
                    {this.props.animal.isGettingTubed ? (<FontAwesomeIcon icon="blender" />): <p></p>}
                </td>
                <td>
                    {this.props.animal.isGettingControlledMeds ? (<FontAwesomeIcon icon="prescription-bottle-alt" />) :
                        <p></p>}
                </td>
            </tr>
        )
    }
}

export default connect()(AnimalTitle)