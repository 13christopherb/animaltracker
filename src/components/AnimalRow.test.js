import React from 'react';
import {shallow} from 'enzyme';
import AnimalTitle from './AnimalRow';
import toJson from 'enzyme-to-json';

const animal = {
    name: 'Kirt',
    species: 'CSL',
    weight: 85,
    isGettingTubed: false,
    isGettingControlledMeds: true,
    id: 3

}

describe('<AnimalRow />', () => {
    describe('render()', () => {
        test('should render the component', () => {
            const wrapper = shallow(<AnimalTitle animal={animal}/>)

            expect(toJson(wrapper)).toMatchSnapshot()
        })
    });
    describe('Buttons', () => {
        test('delete button calles deleteAnimal', () => {
            const mockDelete = jest.fn();
            const wrapper = shallow(
                <AnimalTitle deleteAnimal={mockDelete} animal={animal}/>
            );

            wrapper.find('#delete').simulate('click', { preventDefault() {} })
            expect(mockDelete).toHaveBeenCalled()
        });
    });
});