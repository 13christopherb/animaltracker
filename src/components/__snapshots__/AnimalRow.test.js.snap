// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AnimalRow /> render() should render the component 1`] = `
<tr
  key="3"
>
  <td>
    CSL
  </td>
  <td>
    Kirt
  </td>
  <td>
    85
  </td>
  <td>
    <p />
  </td>
  <td>
    <FontAwesomeIcon
      border={false}
      className=""
      fixedWidth={false}
      flip={null}
      icon="prescription-bottle-alt"
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      symbol={false}
      title=""
      transform={null}
    />
  </td>
  <td>
    <button
      className="btn btn-danger"
      id="delete"
      onClick={[Function]}
    >
      Delete
    </button>
  </td>
</tr>
`;
